dev/io.openliberty.data.internal_fat_ddlgen/test-applications/DDLGenTestApp/src/test/jakarta/data/ddlgen/web/DDLGenTestServlet.java

```diff
@@ -54,29 +82,171 @@ public class DDLGenTestServlet extends FATServlet {
 
-        DataSource ds = InitialContext.doLookup(usingDataSource);
+        DataSource admin = InitialContext.doLookup(usingDataSource);
 
         for (String file : files) {
-            if (!file.contains(withDatabaseStore))
-                continue;
-
             try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
                 System.out.println("Execute DDL file: " + file);
                 String line;
                 while ((line = reader.readLine()) != null) {
                     if (line.isBlank() || line.equals("EXIT;"))
                         continue;
 
-                    System.out.println("  Execute SQL Statement: " + line);
-                    try (Connection con = ds.getConnection(); Statement stmt = con.createStatement()) {
+                    try (Connection con = admin.getConnection(); Statement stmt = con.createStatement()) {
+                        if (!preamble.isBlank()) {
+                            System.out.println("  Execute Preamble: " + preamble);
+                            stmt.executeUpdate(preamble);
+                        }
+                        System.out.println("  Execute SQL Statement: " + line);
                         stmt.execute(line);
                     }
                 }
             }
         }
+
+    }
+
+    /**
+     * Attempt to insert, find, and delete a row in the Cars table
+     * which was created via execution of generated ddl files
+     */
+    @Test
+    public void testSaveToDefaultDatabase() {
+        assertEquals("Table car should not have any starting values", 0, cars.findAll().count());
+
+        String id = cars.save(Car.of("1234", "Honda", "Civic", 2014, 89452, 7500)).vin;
+
+        Car result = cars.findById(id).orElseThrow();
+        assertEquals("Honda", result.make);
+        assertEquals("Civic", result.model);
+        assertEquals(2014, result.modelYear);
+        assertEquals(89452, result.odometer);
+        assertEquals(7500, result.price, 0.1);
+
+        cars.delete(result);
+    }
+
+    /**
+     * Attempt to insert, find, and delete a row in the Trucks table
+     * which was created via execution of generated ddl files
+     */
+    @Test
+    public void testSaveToDatasourceId() throws Exception {
+        assertEquals("Table truck should not have any starting values", 0, trucks.findAll().count());
+
+        String id = trucks.save(Truck.of("1234", "Honda", "Ridgeline", 2025, 10, 40150, 63.6)).vin;
+
+        //Ensure no data was not put into the automobile table
+        DataSource testDataSource = InitialContext.doLookup("jdbc/TestDataSource");
+        try (Connection con = testDataSource.getConnection(); Statement stmt = con.createStatement()) {
+            try (ResultSet rs = stmt.executeQuery("SELECT * FROM Automobile")) {
+                assertFalse("Truck element should have been saved to Truck table, not Automobile", rs.next()); //No data in table
+            }
+        }
```

Comment:
Could use a Resource Accessor Method to obtain the Connection or DataSource from the repository.

Reply:
👍🏼 Good suggestion.  I figured I was already looking up the admin datasource to create the tables.  So our tests could do the same for validation.